src/jenkins.ts

```diff
@@ -0,0 +1,202 @@
+        labels: labels,
+        name: 'cdk8s-jenkins-construct',
+      },
+      spec: {
+        groovyScripts: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        configurationAsCode: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        jenkinsApiSettings: {
+          authorizationStrategy: authorizationStrategy,
+        },
+        master: {
+          disableCsrfProtection: disableCSRFProtection,
+          containers: containers,
+          basePlugins: basePlugins,
+          plugins: plugins,
+        },
+        seedJobs: seedJobs,
+      },
+    });
+
+    this.node.defaultChild = jenkinsCustomResource;
+  }
+
+  public addBasePlugins(basePluginsUpdates: Plugins[]): void {
+    const jenkinsApiObject = ApiObject.of(this);
+
+    basePluginsUpdates.forEach((plugin) => {
+      jenkinsApiObject.addJsonPatch(JsonPatch.add('/spec/master/basePlugins/-', plugin));
+    });
+  }
+
+  public addPlugins(plugins: Plugins[]): void {
+    const jenkinsApiObject = ApiObject.of(this);
+
+    plugins.forEach((plugin) => {
+      jenkinsApiObject.addJsonPatch(JsonPatch.add('/spec/master/plugins/-', plugin));
+    });
+  }
+
+  public addSeedJobs(seedJobs: SeedJobs[]): void {
+    const jenkinsApiObject = ApiObject.of(this);
+
+    seedJobs.forEach((seedJob) => {
+      jenkinsApiObject.addJsonPatch(JsonPatch.add('0/spec/seedJobs/-', seedJob));
+    });
+  }
+}
+
+function getJenkinsBasePlugins(basePluginsUpdates: Plugins[] | undefined): Plugins[] {
+  const basePlugins = [
+    {
+      name: 'kubernetes',
+      version: '1.31.3',
+    },
+    {
+      name: 'workflow-job',
+      version: '1145.v7f2433caa07f',
+    },
+    {
+      name: 'workflow-aggregator',
+      version: '2.6',
+    },
+    {
+      name: 'git',
+      version: '4.10.3',
+    },
+    {
+      name: 'job-dsl',
+      version: '1.78.1',
+    },
+    {
+      name: 'configuration-as-code',
+      version: '1414.v878271fc496f',
+    },
+    {
+      name: 'kubernetes-credentials-provider',
+      version: '0.20',
+    },
+  ];
+
+  if (basePluginsUpdates == undefined || basePluginsUpdates.length == 0) {
+    return basePlugins;
+  }
+
+  basePluginsUpdates.forEach((basePluginUpdate) => {
+    basePlugins.forEach((basePlugin) => {
+      if (basePlugin.name == basePluginUpdate.name && basePlugin.version != basePluginUpdate.version) {
+        basePlugin.version = basePluginUpdate.version;
+      }
+    });
+  });
+
+  return basePlugins;
+}
+
+function getContainerConfiguration(): JenkinsContainerConfiguration[] {
+  let jenkinsContainersConfiguration: JenkinsContainerConfiguration[] = [
+    {
+      name: 'jenkins-master',
+      image: 'jenkins/jenkins:lts',
+      imagePullPolicy: 'Always',
+      resources: {
+        limits: {
+          cpu: JenkinsContainerLimits.fromString('2'),
+          memory: JenkinsContainerLimits.fromString('3Gi'),
```

Comment:
Just curious, are these recommended defaults, or numbers you picked out, or...? Maybe add a comment to add more context 🙂

Reply:
I had added the ones that I was using locally but will add the defaults now which are mentioned [here](https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/installing-the-operator/) :)